mypyc/genops.py

```diff
@@ -595,6 +595,8 @@ def prepare_class_def(path: str, module_name: str, cdef: ClassDef,
                 ir.inherits_python = True
             continue
         base_ir = mapper.type_to_ir[cls]
+        if not base_ir.is_ext_class:
+            ir.inherits_python = True
```

Comment:
Shouldn't this be "transitive"? I.e what if all `base_ir` are all extension classes, but some inherit from Python? I would add a test for this (like add a `class F(E): pass` and check it has `__dict__`).

Reply:
This is already searching the whole mro, but there is a check that isn't part of that search that I'll move in.